app/src/lib/git/branch.ts

```diff
@@ -100,10 +98,18 @@ export async function deleteBranch(
 
     // If the user is not authenticated, the push is going to fail
     // Let this propagate and leave it to the caller to handle
-    const result = await git(args, repository.path, 'deleteRemoteBranch', {
-      env: await envForRemoteOperation(account, remoteUrl),
-      expectedErrors: new Set<DugiteError>([DugiteError.BranchDeletionFailed]),
-    })
+    const result = await withTrampolineEnvForRemoteOperation(
```

Comment:
All these files under `app/src/lib/git` just change a bit because of a design decision: just like we do in other parts of the app (`withPushPullFetch` or `withIsCommitting`), `withTrampolineEnvForRemoteOperation` will wrap a git operation that might need to use the trampoline. The purpose of `withTrampolineEnvForRemoteOperation` is to generate a trampoline token before invoking git, and revoking it right after that.